### PR TITLE
odoo: init at 15.0.20211029

### PR DESCRIPTION
**Pull request by mkg20001**
###### Motivation for this change

Add odoo module

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/finance/odoo.nix [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        default = pkgs.odoo;
+        defaultText = literalExpression "pkgs.odoo";
+        description = "Odoo package to use.";
+      };
+
+      addons = mkOption {
+        type = with types; listOf package;
+        default = [];
+        example = literalExpression "[ pkgs.odoo_enterprise ]";
+        description = "Odoo addons";
```
Comment (aanderse) [inline nixos/modules/services/finance/odoo.nix:25]: `description` values end with a period.

File: nixos/modules/services/finance/odoo.nix [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = {};
+        description = ''
+          Odoo configuration settings. For more details see https://www.odoo.com/documentation/15.0/administration/install/deploy.html
```
Comment (aanderse) [inline nixos/modules/services/finance/odoo.nix:32]: Would be nice to `xlink` that.

File: nixos/modules/services/finance/odoo.nix [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    systemd.services.odoo = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" "postgresql.service" ];
+
+      # pg_dump
+      path = [ config.services.postgresql.package ];
+
+      requires = [ "postgresql.service" ];
+      script = "HOME=$STATE_DIRECTORY ${cfg.package}/bin/odoo ${optionalString (cfg.addons != []) "--addons-path=${concatMapStringsSep "," escapeShellArg cfg.addons}"} -c ${cfgFile}";
```
Comment (aanderse) [inline nixos/modules/services/finance/odoo.nix:103]: You should set `HOME` via `environment.HOME` and then you can use `ExecStart` instead. Less indirection.

**Review by aanderse — COMMENTED** [review]
Sorry I meant to review this earlier. No big issues, but if you find the time there are a couple minor points you may or may not want to address.

File: nixos/modules/services/finance/odoo.nix (hunk 2 above)
Comment (mkg20001) [reply nixos/modules/services/finance/odoo.nix:32]: how's that work?

File: nixos/modules/services/finance/odoo.nix (hunk 3 above)
Comment (mkg20001) [reply nixos/modules/services/finance/odoo.nix:103]: Does that take variables?

File: nixos/modules/services/finance/odoo.nix (hunk 1 above)
Comment (mkg20001) [reply nixos/modules/services/finance/odoo.nix:25]: I've been doing it wrong this whole time

File: nixos/modules/services/finance/odoo.nix (hunk 2 above)
Comment (aanderse) [reply nixos/modules/services/finance/odoo.nix:32]:
```suggestion
          Odoo configuration settings. For more details see <link xlink:href="https://www.odoo.com/documentation/15.0/administration/install/deploy.html"/>
```

Makes nice clicky in the manual and search.nixos.org :heart:

File: nixos/modules/services/finance/odoo.nix (hunk 3 above)
Comment (aanderse) [reply nixos/modules/services/finance/odoo.nix:103]: No :disappointed: In this case you know what it is, though.

**Pull request merged**